api/src/db/connect.ts

```
@@ -0,0 +1,26 @@
+// import knex from 'knex'
+// const pg = knex({
+//   client: 'pg',
+//   connection: {
+//     host : '127.0.0.1',
+//     port : 2137,
+//     user : 'postgres', //ENV
+//     password : 'password', //ENV
+//     database : 'postgres' //ENV?
+//   }
+// });
+
+const knex = require('knex')(
+  {
+    client: 'pg',
+    connection: {
+      host : '127.0.0.1',
```

If you want to define the way of retrieving values later - you can make a class which will provide this values.
Then you can change the definition of value providing class (or return different implementation) changing only one place.
Read about https://www.typescriptlang.org/docs/handbook/interfaces.html if you want about separating contract from implementation.

---

For example (to make it clearer):
You can create file called 'env.ts' that first will export the hardcoded structure, and then you may change it to structure that will return value from environment defaulting to hardcoded structure -> see delegation and decorator pattern if you want to learn them on this example.